pkg/arm/v15/data/master-startup.sh

```diff
@@ -74,10 +74,6 @@ done
 
 iptables-save >/etc/sysconfig/iptables
 
-# when starting node waagent and network utilities goes into race condition.
-# if waagent runs before dns is known to the node we end up with empty string
-while [[ $(hostname -d) == "" ]]; do sleep 1; done
```

Comment:
Why do we move this after image pulls?

Reply:
When setting custom dns on the private link vnet the internal Azure naming is no longer given to the cluster.  This means that we will never pass the above `hostname -d` check.  How we solve this is to query for the Azure dns domain inside of the startup container and write the settings to the correct files.  Then check for the hostname afterwards.